:tada: show explorers and incoming sdg references in the charts admin

danyx23 opened the pull request:
This PR uses the fact that we now have up-to-date information in our DB of which explorers use which charts (by id). The way this information is used is in the chart admin in the references panel.

To complete the picture, this PR also adds a new table into the DB that stores the chart slugs and ids that are referenced from the [sdg-tracker.org](https://sdg-tracker.org) website and shows this in the references section as well.

sophiamersmann commented on adminSiteServer/apiRouter.ts:
```diff
@@ -173,39 +207,45 @@ const getReferencesByChartId = async (
     // `grapher/gdp` is a substring of `grapher/gdp-maddison`. We need to be
     // careful not to erroneously match those, which is why we switched to a
     // REGEXP.
```
This comment is now misplaced, right?

sophiamersmann reviewed (approved):
Great to have this information in the admin! 🙌🏻 

(I reviewed faithfully but didn't really understand the migration but I'm sure Marcel will)

marcelgerber commented on adminSiteClient/EditorReferencesTab.tsx:
```diff
@@ -1,13 +1,139 @@
+        <>
+            <React.Fragment>
+                <p>
+                    Legacy SDG referenced charts (slugs of charts only, links to
+                    precise SDG page are not available):
```
I can generate a list of slugs together with the article they appear, shouldn't be too much work.

danyx23 replied:
I don't think this is needed at this point - we'll migrate away soon anyhow

marcelgerber commented on adminSiteClient/EditorReferencesTab.tsx:
```diff
@@ -1,13 +1,139 @@
+                <ul className="list-group">
+                    {props.references.legacySdgCharts.map((sdgSlug) => (
+                        <li key={sdgSlug} className="list-group-item">
+                            sdgSlug
```
```suggestion
                            {sdgSlug}
```

danyx23 commented on db/migration/1685706058408-AddLegacySdgChartReferences.ts:
```diff
@@ -0,0 +1,597 @@
+                console.error(`No chart found for ${slug}`)
+                continue
+            }
+            if (rows.length > 1)
+                console.error(
+                    `Multiple charts found for ${slug} - selecting the first one`
+                )
```
Yeah good point - added the check for `publishedBy is not null` and extracted a function

danyx23 commented:
@marcelgerber could I have your eyes over this once more after I addressed the comments? I rebased on top of master so the broken staging is because it was created with node 16.
